index.js

```diff
@@ -290,19 +322,13 @@ export default class OneSignal {
     static setEmail(email, emailAuthCode, callback) {
         if (!checkIfInitialized()) return;
 
-        if (emailAuthCode == undefined) {
-            //emailAuthCode is an optional parameter
-            //since JS does not support function overloading,
-            //unauthenticated setEmail calls will have emailAuthCode as the callback
+        if (emailAuthCode === undefined)
+            emailAuthCode = '';
 
-            RNOneSignal.setUnauthenticatedEmail(email, function(){});
-        } else if (callback == undefined && typeof emailAuthCode == 'function') {
-            RNOneSignal.setUnauthenticatedEmail(email, emailAuthCode);
-        } else if (callback == undefined) {
-            RNOneSignal.setEmail(email, emailAuthCode, function(){});
-        } else {
-            RNOneSignal.setEmail(email, emailAuthCode, callback);
-        }
+        if (callback === undefined)
+            callback = function(){};
```

Comment:
Do we need to do this? Does sending `null` or `undefined` to `RNOneSignal.setEmail` issues?

Reply:
Can I explain on a call?

Reply:
I think we forgot to talk about this one. But if it is needed to prevent an issue we can leave it.